src/renderer/root/components/AuthenticatedLayout/AddressBar/AddressBar.js

```diff
@@ -92,14 +93,24 @@ export default class AddressBar extends React.PureComponent {
         onClick={this.props.onToggleSidebar}
       />
     );
-  }
+  };
 
   handleKeyDown = (event) => {
     if (event.which === RETURN_KEY) {
       this.props.onQuery(event.target.value);
     }
   }
 
+  handleSelect = () => {
+    const selection = window.getSelection ? window.getSelection() : null;
+    if (isEmpty(selection.toString())) {
+      this.input.select();
+    } else {
+      selection.empty();
+      this.input.select();
```

Comment:
You're calling `this.input.select()` in both branches.  I think you could simplify this down to:

```js
if (!isEmpty(selection.toString())) {
  selection.empty();
}

this.input.select();
```